src/main/java/com/hierynomus/sshj/transport/verification/KnownHostMatchers.java

```diff
@@ -80,17 +86,22 @@ private static class HashedHostMatcher implements HostMatcher {
 
         @Override
         public boolean match(String hostname) throws IOException {
-            return hash.equals(hashHost(hostname));
+            try {
+                return hash.equals(hashHost(hostname));
+            } catch (Base64DecodeError err) {
+                log.warn("Failed to match {}", hostname, err);
```

Comment:
It seems like this error should indicate that salt decoding failed as the cause of matching failures.
```suggestion
                log.warn("Hostname [{}] not matched: salt decoding failed", hostname, err);
```

Reply:
https://github.com/hierynomus/sshj/pull/936/commits/1df0e48bced96f767456f4d203bf5e36ba2be01e